File: Mid/grade.java
import java.io.*;

class grade{

   public static void main(String[] args) throws IOException{ 

         String text = "", rules = "", ruletext, temp;
         int currgrade = 0, points = 0, i = 0;
     
         //get student code and store to variable 'text
         text = args[0];   
          
         //store rules into an array
         String[] rulearray = {args[1],args[2],args[3],args[4]};
         
         //grading
         while (i < rulearray.length){
              ruletext = rulearray[i];
              points = grade(ruletext, text);
              currgrade = currgrade + points;
              points = 0;
              i++; 
          } 
          
          //25 points max per question
          if(currgrade > 25){
              currgrade = 25;
          }
        
          System.out.println(currgrade);  
     
  }
  
     public static int grade(String x, String text){
     int count = 0;
     
     if(text.contains(x)){        
                count = count + 25;      
            } else{       
                //do nothing            
            }       
           return count;         
          }  

} 

File: Mid/grade.py
import sys

student_text, ruletext = "", ""
currgrade, points, i = 0, 0, 0

#get student code and assign it to 'text'
student_text = sys.argv[1]
  
#store rules into array
rulearray = [sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5]]

#function to check if method is within student input
def grade(rules, student_text):
    count = 0
    
    if rules in student_text:
      count = 5
      return count
    else:
      count = 0
      return count

while(i < len(rulearray)):
    ruletext = rulearray[i];
    points = grade(ruletext, student_text)
    currgrade = currgrade + points
    points = 0
    i += 1
 
if(currgrade > 5):
    currgrade = 5
  
print(currgrade)
  
